CRM/Mailing/Event/BAO/TrackableURLOpen.php

```diff
@@ -364,9 +364,9 @@ public static function &getRows(
       //Added "||$rowCount" to avoid displaying all records on first page
       $query .= ' LIMIT ' . CRM_Utils_Type::escape($offset, 'Integer') . ', ' . CRM_Utils_Type::escape($rowCount, 'Integer');
     }
-
+    CRM_Core_DAO::disableFullGroupByMode();
     $dao->query($query);
-
+    CRM_Core_DAO::enableFullGroupByMode();
```

Comment:
> I'm happy if jenkins is

Agree and disagree with this comment.

The idea of temporarily disabling `ONLY_FULL_GROUP_BY` seems good. (I originally voted to disable it across all of Civi... so doing it in a narrower way is also OK.)

The implementation in this PR feels a little a leaky -- e.g. if the system starts in
 `ONLY_FULL_GROUP_BY`=on, then this correctly restores it. However, if the system starts without `ONLY_FULL_GROUP_BY`=off, then this incorrectly switches us to `ONLY_FULL_GROUP_BY`=on. A leak like this can influence other code.

In practice, `TrackableURLOpen` is generally called on its own (without a lot of other business logic), so a leak isn't likely to be symptomatic. The main time we care about leaks is in running unit-tests. Thus, I mostly agree with @eileen's point -- if it works in the tests, then it's OK.

*However*, `TrackableURLOpen` is a bit special in how very isolated the use-case is. If you were to reproduce this coding pattern in other contexts, then there'd be more risk of a leak producing symptoms.

Which is a long-winded way of saying... this passes and it's OK, but when making a temporary change to global state in the future, let's try to be a little more strict about cleanup.

Reply:
@totten will it turn it back on tho? i think by the inclusion of the checking of this function https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/SQL.php#L114 in the turning back on function will mean that ONLY if its in the global defaults and its not present at the moment will the ONLY_FULL_GROUP_BY mode be added back to the sessions sql_modes

Reply:
@seamuslee001 @totten should the function name be changed to reenableFullGroupByModeIfConfigured or something to better match what it does?

Reply:
I would be fine with that

Reply:
It's worth getting this right because I believe this approach should be our first line of defence against breakage in full group by mode & once we stop seeing new ones we should start to slowly & with many tests tackle the root cause of the queries. We have had too many resulting breakages up until now from full group by fixes for me to be comfortable with fixing the queries largely in isolation.

Reply:
@seamuslee001 but don't you hate the way re-enable doesn't camel well?

Reply:
Oh, good call in pointing circa line 114. I didn't realize that it was based on the default (aka `@@global.sql_mode`). Agree it's not a functional problem and just a misinterpretation of the function's intent -- `reenableFullGroupByModeIfConfigured()` does seem more astute description.